dns-controller/pkg/watchers/node.go

```diff
@@ -220,9 +221,11 @@ func (c *NodeController) updateNodeRecords(node *v1.Node) string {
 	// node/role=<role>/external -> ExternalIP
 	// node/role=<role>/internal -> InternalIP
 	{
-		role := kopsutil.GetNodeRole(node)
-		// Default to node
-		if role == "" {
+		role := kopsutil.GetNodeRoles(node)
```

Comment:
I think we should have `GetNodeRoles` return []string.  If the list had `len() == 0` then we default to `[]string{ "node" }` as today.

Then we can loop over each of the roles, so that the alias will work for nodes with multiple roles.